router/handler_ctrl/route.go

```diff
@@ -156,13 +161,30 @@ func (rh *routeHandler) connectEgress(msg *channel.Message, attempt int, ch chan
 				if peerData, err := dialer.Dial(route.Egress.Destination, circuitId, xgress.Address(route.Egress.Address), bindHandler, ctx); err == nil {
 					rh.success(msg, attempt, route, peerData, log)
 				} else {
-					rh.fail(msg, attempt, route, errors.Wrapf(err, "error creating route for [c/%s]", route.CircuitId), log)
+					var headerError *byte
+					if _, ok := err.(xgress.InvalidTerminatorError); ok {
+						var errCode byte = ctrl_msg.ErrorTypeInvalidTerminator
+						headerError = &errCode
+						switch {
+						case errors.Is(err, syscall.ECONNREFUSED):
+							errCode = ctrl_msg.ErrorTypeConnectionRefused
+						case errors.Is(err, syscall.ETIMEDOUT):
+							errCode = ctrl_msg.ErrorTypeDialTimedOut
+						case errors.As(err, &xgress.MisconfiguredTerminatorError{}):
+							errCode = ctrl_msg.ErrorTypeMisconfiguredTerminator
+						case errors.As(err, &xgress.InvalidTerminatorError{}):
+							errCode = ctrl_msg.ErrorTypeInvalidTerminator
+						}
+						headerError = &errCode
+					}
+					rh.fail(msg, attempt, route, errors.Wrapf(err, "error creating route for [c/%s]", route.CircuitId), headerError, log)
 				}
 			} else {
-				rh.fail(msg, attempt, route, errors.Wrapf(err, "unable to create dialer for [c/%s]", route.CircuitId), log)
+				var errCode byte = ctrl_msg.ErrorTypeMisconfiguredTerminator
+				rh.fail(msg, attempt, route, errors.Wrapf(err, "unable to create dialer for [c/%s]", route.CircuitId), &errCode, log)
 			}
 		} else {
-			rh.fail(msg, attempt, route, errors.Wrapf(err, "error creating route for [c/%s]", route.CircuitId), log)
+			rh.fail(msg, attempt, route, errors.Wrapf(err, "error creating route for [c/%s]", route.CircuitId), nil, log)
```

Comment:
This should also be misconfigured terminator

Reply:
Added. I don't exactly understand this one though. This would happen when a factory is needed that doesn't exist? I'm probably just thinking about it too much, but I don't see why that would be a terminator misconfiguration

Reply:
This would happen if the user created with a terminator with binding 'edge_tansplort' instead of 'edge_transport'. There would be no xgress Factory registered for that binding and you'd get that first error. But they could fix it by fixing the terminator binding.